test/e2e/network/netpol/kubemanager.go

```diff
@@ -151,89 +145,27 @@ func (k *kubeManager) getPod(ctx context.Context, ns string, name string) (*v1.P
+		negative, strings.ToLower(string(args.protocol)), net.JoinHostPort(args.addrTo, port))}
 
 	// NOTE: The return value of this function[probeConnectivity] should be true if the probe is successful and false otherwise.
-
-	// probeError will be the return value of this function[probeConnectivity] call.
-	var probeError error
-	var stderr string
-
-	// Instead of re-running the job on connectivity failure, the following conditionFunc when passed to PollImmediate, reruns
-	// the job when the observed value don't match the expected value, so we don't rely on return value of PollImmediate, we
-	// simply discard it and use probeError, defined outside scope of conditionFunc, for returning the result of probeConnectivity.
-	conditionFunc := func() (bool, error) {
-		_, stderr, probeError = k.executeRemoteCommand(args.nsFrom, args.podFrom, args.containerFrom, cmd)
-		// retry should only occur if expected and observed value don't match.
-		if args.expectConnectivity {
-			if probeError != nil {
-				// since we expect connectivity here, we fail the condition for PollImmediate to reattempt the probe.
-				// this happens in the cases where network is congested, we don't have any policy rejecting traffic
-				// from "podFrom" to "podTo" and probes from "podFrom" to "podTo" are failing.
-				framework.Logf("probe #%d :: connectivity expected :: %s/%s -> %s :: stderr - %s",
-					attempt+1, args.nsFrom, args.podFrom, args.addrTo, stderr,
-				)
-				attempt++
-				return false, nil
-			} else {
-				// we got the expected results, exit immediately.
-				return true, nil
-			}
-		} else {
-			if probeError != nil {
-				// we got the expected results, exit immediately.
-				return true, nil
-			} else {
-				// since we don't expect connectivity here, we fail the condition for PollImmediate to reattempt the probe.
-				// this happens in the cases where we have policy rejecting traffic from "podFrom" to "podTo", but CNI takes
-				// time to implement the policy and probe from "podFrom" to "podTo" was successful in that window.
-				framework.Logf(" probe #%d :: connectivity not expected :: %s/%s -> %s",
-					attempt+1, args.nsFrom, args.podFrom, args.addrTo,
-				)
-				attempt++
-				return false, nil
-			}
-		}
-	}
-
-	// ignore the result of PollImmediate, we are only concerned with probeError.
-	_ = wait.PollImmediate(
-		time.Duration(args.pollIntervalSeconds)*time.Second,
-		time.Duration(args.pollTimeoutSeconds)*time.Second,
-		conditionFunc,
-	)
-
-	if probeError != nil {
-		return false, commandDebugString, nil
-	}
-	return true, commandDebugString, nil
+	_, stderr, probeError := k.executeRemoteCommand(args.nsFrom, args.podFrom, args.containerFrom, cmd)
+	framework.Logf("probe %s/%s -> %s took %v expected %v :: stderr - %s probeError - %v", args.nsFrom, args.podFrom, args.addrTo, time.Since(now), args.expectConnectivity, stderr, probeError)
```

Comment:
could make this better by splitting it into different messages depending on expectation and result.

(Also, `time.Since(now)` looks silly. Should probably have called that `start`.)